File: algo-2/latihan/bab-3/latihan3_1_4.py
print("Program Menghitung Luas")
print("***********************")
print("Pilih Menu")
print("=====> 1. Luas Lingkaran")
print("       2. Luas Persegi")
print("       3. Luas Segitiga")

pilihan = int(input("Pilih: "))

print()
if pilihan == 1:
    print("Program Lingkaran")
    print("*****************")

    radius = int(input("Masukkan jari-jari: "))

    luas = 3.14 * radius * radius

    print("Luas lingkaran =", luas)
elif pilihan == 2:
    print("Program Persegi Panjang")
    print("***********************")

    panjang = int(input("Masukkan panjang: "))
    lebar = int(input("Masukkan lebar: "))

    luas = panjang * lebar

    print("Luas persegi panjang =", luas)
elif pilihan == 3:
    print("Program Segitiga")
    print("****************")

    alas = int(input("Masukkan alas: "))
    tinggi = int(input("Masukkan tinggi: "))

    luas = 0.5 * alas * tinggi

    print("Luas Segitiga =", luas)
else:
    print("Pilihan menu tidak ada")

File: algo-2/latihan-haskell/bab-3/latihan3_1_4.hs
main :: IO ()
main = do
  putStrLn "Program Menghitung Luas"
  putStrLn "***********************"
  putStrLn "Pilih Menu"
  putStrLn "=====> 1. Luas Lingkaran"
  putStrLn "       2. Luas Persegi"
  putStrLn "       3. Luas Segitiga"

  putStr "Pilih: "
  menu <- readLn

  putStr "\n"

  case menu of
    1 -> menuLuasLingkaran
    2 -> menuLuasPersegiPanjang
    3 -> menuLuasSegitiga
    _ -> putStrLn "Pilihan menu tidak ada"

menuLuasLingkaran :: IO ()
menuLuasLingkaran = do
  putStrLn "Program Lingkaran"
  putStrLn "*****************"

  putStr "Masukkan jari-jari: "
  radius <- readLn

  let luas = 3.14 * radius * radius
   in putStrLn ("Luas Lingkaran = " ++ show luas)

menuLuasPersegiPanjang :: IO ()
menuLuasPersegiPanjang = do
  putStrLn "Program Persegi Panjang"
  putStrLn "***********************"

  putStr "Masukkan panjang: "
  panjang <- readLn
  putStr "Masukkan lebar: "
  lebar <- readLn

  let luas = panjang * lebar
   in putStrLn ("Luas Persegi Panjang = " ++ show luas)

menuLuasSegitiga :: IO ()
menuLuasSegitiga = do
  putStrLn "Program Segitiga"
  putStrLn "****************"

  putStr "Masukkan alas: "
  alas <- readLn
  putStr "Masukkan tinggi: "
  tinggi <- readLn

  let luas = 0.5 * alas * tinggi
   in putStrLn ("Luas Segitiga = " ++ show luas)
